src/physics/force_density.jl

```diff
@@ -20,8 +20,56 @@ function _calc_force_density!(storage::AbstractStorage, system::AbstractSystem,
                               params::AbstractPointParameters, each_point_idx)
     storage.b_int .= 0
     storage.n_active_bonds .= 0
-    for point_id in each_point_idx
+    Threads.@threads :static for point_id in each_point_idx
         force_density_point!(storage, system, mat, params, point_id)
     end
     return nothing
 end
+
+function calc_force_density!(b::AbstractBodyChunk, nhs)
+    _calc_force_density!(b.storage, b.system, b.mat, b.paramsetup, each_point_idx(b), nhs)
+    return nothing
+end
+
+function calc_force_density!(b::AbstractBodyChunk, nhs::Nothing)
+    # Call original function
+    _calc_force_density!(b.storage, b.system, b.mat, b.paramsetup, each_point_idx(b))
+    return nothing
+end
+
+function _calc_force_density!(storage::AbstractStorage, system::AbstractSystem,
+                              mat::AbstractMaterial,
+                              params::AbstractPointParameters, each_point_idx, nhs)
+    storage.b_int .= 0
+    storage.n_active_bonds .= 0
+
+    # for_particle_neighbor(storage.position, storage.position, nhs,
+    #                       particles=each_point_idx, parallel=false) do i, j, _, L
+    Threads.@threads :static for point_id in each_point_idx
+    foreach_neighbor(storage.position, storage.position, nhs, point_id) do i, j, _, L
```

Comment:
`storage.position` is the current position, not the initial position, right? But `L` is supposed to be the initial distance, right?

Then I have to pass the initial position here to PointNeighbors.jl, but I didn't find where it's stored. This might change the benchmark slightly, as another array has to be loaded into the cache.

Reply:
Yes, `storage.position` is the current position, and `system.position` the initial position. The initial length of a bond `L` is stored in `bond.length`.